### PR TITLE
banshee: Add support for fmv instructions

### PR DESCRIPTION
**Pull request by noah95**
Next attempt!

This implements `fmv.x.w` as a float store and `fmv.w.x` as an integer store. Both work as tested in the newly created test `fmv.c` which was compiled with llvm.

Open points
 - [x] Do I have to specify how the new unit test is built? Or are the source/dump/binary simply checked in
 - [x] Did I understand the behavior of `fmv` correctly? Are the upper 32-bit of the floating point register left as-is or modified, and how is the behavior of an `LLVMBuildStore` in this case?

### CODE REVIEW

File: sw/banshee/src/tran.rs [hunk 1]
```diff
@@ -1658,6 +1658,51 @@ impl<'a> InstructionTranslator<'a> {
...
+                self.trace_access(
+                    TraceAccess::WriteFReg(data.rd as u8),
+                    LLVMBuildLoad(self.builder, raw_ptr, NONAME),
+                );
+                return Ok(());
+            }
+            _ => (),
+        }
+
+        // Handle DMA operations
```
Comment (fabianschuiki) [inline sw/banshee/src/tran.rs:1705]: Nice solution!

File: sw/banshee/tests/fmv.c [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+        "fmv.w.x %1, x0\n"  // float = 0
+        "fmv.w.x %1, %0\n"  // float <- int
+
+        "mv      %0, x0\n"  // int   = 0
+        "fmv.x.w %0, %1\n"  // int   <- float
+
+        : "+r"(d_rd), "+f"(f_rd));
+
+    return 0;
+}
```
Comment (fabianschuiki) [inline sw/banshee/tests/fmv.c:21]: Neat!

**Review by fabianschuiki — APPROVED** [review]
Looks very nice to me! We've decided to check in the compile RISC-V binaries for unit and integration tests to make it much simpler to cross-check modifications to Banshee without needing a full compiler stack all setup. So :+1: for including them here :smiley: .

**Comment by fabianschuiki** [comment]
Ready to merge when you are :+1:

**Comment by fabianschuiki** [comment]
One minor comment: You might want to add the fmv test to `sw/banshee/tests/Makefile`, just in case :smiley:

**Comment by noah95** [comment]
> One minor comment: You might want to add the fmv test to `sw/banshee/tests/Makefile`, just in case smiley

Done. I will look into the behavior of the upper 32-bits later today

**Comment by noah95** [comment]
from rtl simulations, this does now do the same so can be merged imo

**Pull request merged**